btree/btree.c

```diff
@@ -3166,21 +3175,186 @@ int64_t btree_destroy_tick(struct m0_sm_op *smop)
 	}
 }
 
-int64_t btree_get_tick(struct m0_sm_op *smop)
+/**
+ *  Search from the leaf + 1 level till the root level and find a node
+ *  which has valid sibling. Once found, get the leftmost leaf record from the
+ *  sibling subtree.
+ */
+int  btree_sibling_first_key_get(struct m0_btree_oimpl *oi, struct td *tree,
+				 struct slot *s)
 {
-	struct m0_btree_op 	*bop = M0_AMB(bop, smop, bo_op);
-	//ToDo: Implement complete destroy tick function.
-	switch(bop->bo_op.o_sm.sm_state)
-	{
-		case P_INIT:
-			return P_ACT;
+	int	        i;
+	struct level   *lev;
+	struct segaddr  child;
+
+	for (i = oi->i_used - 1; i >= 0; i--) {
+		lev = &oi->i_level[i];
+		if (lev->l_idx < node_count(lev->l_node)) {
```

Comment:
This check might need to be:
if (lev->l_idx < (node_count(lev->l_node)-1))

Reply:
This check is for leaf nodes. Please recheck node_count and node_find function.

Reply:
The check is correct. Please check `node_count` and `node_find` function.